src/com/amazon/ion/impl/lite/IonSystemLite.java

```diff
@@ -317,156 +317,19 @@ public SymbolTable newSharedSymbolTable(String name,
 
     public IonValueLite newValue(IonReader reader)
     {
-        IonValueLite value = load_value_helper(reader, /*isTopLevel*/ true);
+        IonValueLite value = load_value_helper(reader);
         if (value == null) {
             throw new IonException("No value available");
         }
         return value;
     }
 
-    private IonValueLite load_value_helper(IonReader reader, boolean isTopLevel)
+    private IonValueLite load_value_helper(IonReader reader)
     {
-        boolean symbol_is_present = false;
-
-        IonType t = reader.getType();
-        if (t == null) {
-            return null;
-        }
-        IonValueLite v;
-        if (reader.isNullValue()) {
-            v = newNull(t);
-        }
-        else {
-            switch (t) {
-            case BOOL:
-                v = newBool(reader.booleanValue());
-                break;
-            case INT:
-                // TODO amazon-ion/ion-java/issues/9  Inefficient since we can't determine the size
-                // of the integer in order to avoid making BigIntegers.
-                v = newInt(reader.bigIntegerValue());
```

Comment:
🗺️ I addressed this `TODO` in my rewrite; the method now calls `getIntegerSize()` to see whether `longValue()` will suffice.